pkg/identityserver/application_registry.go

```diff
@@ -333,6 +340,29 @@ func (is *IdentityServer) purgeApplication(ctx context.Context, ids *ttnpb.Appli
 	return ttnpb.Empty, nil
 }
 
+func (is *IdentityServer) issueDevEUI(ctx context.Context, ids *ttnpb.ApplicationIdentifiers) (*ttnpb.IssueDevEUIResponse, error) {
+	if err := rights.RequireApplication(store.WithSoftDeleted(ctx, false), *ids, ttnpb.RIGHT_APPLICATION_DEVICES_WRITE); err != nil {
+		return nil, err
+	}
+	if !is.config.DevEUIBlock.Enabled {
+		return nil, errDevEUIIssuingNotEnabled.New()
+	}
+	res := &ttnpb.IssueDevEUIResponse{}
+	err := is.withDatabase(ctx, func(db *gorm.DB) error {
```

Comment:
Use a transaction here too - you shouldn't need the spin loop anymore I think, if you do.